S3/S3.py

```diff
@@ -701,6 +704,16 @@ def object_put(self, filename, uri, extra_headers = None, extra_label = ""):
             headers['x-amz-server-side-encryption'] = 'aws:kms'
             headers['x-amz-server-side-encryption-aws-kms-key-id'] = self.config.kms_key
 
+        if self.config.sse_customer_key:
+            md5s = md5()
+            sse_customer_key = self.config.sse_customer_key.encode()
+            md5s.update(sse_customer_key)
+            md5_encoded = b64encode(md5s.digest())
+            key_encoded = b64encode(sse_customer_key)
+            headers["x-amz-server-side-encryption-customer-algorithm"] = "AES256"
+            headers["x-amz-server-side-encryption-customer-key"] = key_encoded.decode()
+            headers["x-amz-server-side-encryption-customer-key-md5"] = md5_encoded.decode()
```

Comment:
I put the comment here, but same for all `decode()` and `encode()`:
Even if the effect will be the same, it would be better to use the `encode_to_s3()` and `decode_from_s3()` instead.